File: my-folder/problems/1537-maximum-score-after-splitting-a-string/solution.py
class Solution:
    def maxScore(self, s: str) -> int:
        
        right = s.count("1")
        left = 0
        ans = 0
        for i in range(len(s)-1):
            if s[i] == '0':
                left += 1
            else:
                right -= 1
            ans = max(ans, left+right)
        
        return ans

File: my-folder/problems/1537-maximum-score-after-splitting-a-string/solution.cpp
class Solution {
public:
    int maxScore(string s) {
        int left = 0;
        int right = 0;
        int last = s.length()-1;
        for(int i=0; i<= last; i++)
        {
            if (s[i] == '1') right++;
        }

        int ans = 0;
        for(int i=0; i<last; i++)
        {
            if (s[i] == '0') left++;
            else right--;
            ans = std::max(ans, left+right);
        }
        return ans;
    }
};
